src/runtime-helpers.js

```diff
@@ -502,16 +533,18 @@ async function deployPackage (entities, ow, logger) {
   let opts = await ow.actions.client.options
   let ns = opts.namespace
   for (let pkg of entities.pkgtoCreate) {
-    let options = {}
-    options['name'] = pkg.name
     logger(`Info: Deploying package [${pkg.name}]...`)
-    await ow.packages.update(options)
+    await ow.packages.update(pkg)
     logger(`Info: package [${pkg.name}] has been successfully deployed.\n`)
   }
   for (let action of entities.actions) {
     if (action['exec'] && action['exec']['kind']) {
       action['exec']['components'] = action['exec']['components'].map(sequence => {
-        return `/${ns}/${sequence}`
+        let actionItemCount = sequence.split('/').length
+        let startsWithSlash = sequence.startsWith('/')
+        return (startsWithSlash && actionItemCount > 3) || (!startsWithSlash && actionItemCount > 2)
```

Comment:
This part of the code is very unclear. perhaps I'm misunderstanding the test inputs, let me know.

If ns = 'mynamespace'

Test sequences:
1. `/one/two/three/four` (true case)
2. `one/two/three` (true case)
3. `one/two` (false case)
4. `/one/two/three` (false case)

Will return:
1. `//one/two/three/four`
2. `/one/two/three`
3. `/mynamespace/one/two`
4. `/mynamespace//one/two/three`

Are these the intended results? Perhaps this should be broken out in its own function and unit-tested just in case.

Reply:
Thx @shazron for the review. You are right about the cases except the fourth which is actually a 'true case'. Here is a simple explanation. If there are three parts in the "action path", the first part is the namespace and we don't have to append namespace to it. I will add some comments in the code and also change the test a little so it's more clear.

Reply:
Ah ok, now I get what was intended. 

Actually I was demonstrating what your code will output, if you input my test sequences. If you look at your `if` statement, the fourth item is false case: startsWithSlash is `true` and actionItemCount is `3`, which will not satisfy any of the clauses.

Reply:
Actually the actionItemCount is 4 in this case. Nevertheless, i have simplified that part of code. Please take a look. 🙂

Reply:
Oops. Ah yes, there is an empty string. My bad :)
```
$ "/one/two/three".split('/')
["", "one", "two", "three"] 
```